machine_learning_hep/optimiser.py

```diff
@@ -391,27 +393,47 @@ def do_significance(self):
         acc, acc_err = calc_eff(numacc, denacc)
         self.logger.debug("Acceptance: %.3e +/- %.3e", acc, acc_err)
         #calculation of the expected fonll signals
-        df_fonll = pd.read_csv(self.f_fonll)
-        ptmin = self.p_binmin
-        ptmax = self.p_binmax
-        df_fonll_in_pt = df_fonll.query('(pt >= @ptmin) and (pt < @ptmax)')[self.p_fonllband]
-        prod_cross = df_fonll_in_pt.sum() * self.p_fragf * 1e-12 / len(df_fonll_in_pt)
-        delta_pt = ptmax - ptmin
-        signal_yield = 2. * prod_cross * delta_pt * self.p_br * acc * self.p_taa \
-                       / (self.p_sigmamb * self.p_fprompt)
-        self.logger.debug("Expected signal yield: %.3e", signal_yield)
-        signal_yield = self.p_raahp * signal_yield
-        self.logger.debug("Expected signal yield x RAA hp: %.3e", signal_yield)
-
-        #now we plot the fonll expectation
-        plt.figure(figsize=(20, 15))
-        plt.subplot(111)
-        plt.plot(df_fonll['pt'], df_fonll[self.p_fonllband] * self.p_fragf, linewidth=4.0)
-        plt.xlabel('P_t [GeV/c]', fontsize=20)
-        plt.ylabel('Cross Section [pb/GeV]', fontsize=20)
-        plt.title("FONLL cross section " + self.p_case, fontsize=20)
-        plt.semilogy()
-        plt.savefig(f'{self.dirmlplot}/FONLL_curve_{self.s_suffix}.png')
+
+        if self.p_isfonll:
+            df_fonll = TFile.Open(self.f_fonll)
+            df_fonll_Lc = df_fonll.Get(self.p_fonllparticle+"pred_"+self.p_fonllband)
+            ptmin = self.p_binmin
+            ptmax = self.p_binmax
```

Comment:
Why do you introduce another variable for this? You could just use `self.p_binmin`/`self.binmax` in the following